app/controllers/data_extraction.py

```diff
@@ -0,0 +1,38 @@
+import os
+from typing import BinaryIO
+
+from pipeline import LabelStorage, analyze
+
+from app.constants import UPLOAD_FOLDER
+
+
+def extract_data(files: dict[str, BinaryIO], ocr, gpt):
```

Comment:
Return type?

Reply:
done. Although this one can be inferred by python from the `analyze` function.